pulsar-io/core/pom.xml

```diff
@@ -47,6 +47,19 @@
         </exclusion>
       </exclusions>
     </dependency>
+    
+    <dependency>
```

Comment:
Why are we adding this dependency here?

Reply:
`EncryptionContext` contains `CompressionType` which is a type of `org.apache.pulsar.shaded.com.google.protobuf.v241.Internal.EnumLite` that presents under `protobuf-shaded`. so, any pulsar-io which wants to access `EncryptionContext` requires this dependency.